src/bin/cargo-flamegraph.rs

```diff
@@ -174,28 +208,31 @@ fn build(opt: &Opt) {
         cmd.arg("--no-default-features");
     }
 
+    cmd.arg("--message-format=json-render-diagnostics");
+
     if opt.verbose {
         println!("build command: {:?}", cmd);
     }
 
-    let mut child = cmd
-        .spawn()
-        .expect("failed to spawn cargo build command");
+    let Output { status, stdout, .. } = cmd
+        .stderr(Stdio::inherit())
+        .output()
+        .expect("failed to execute cargo build command");
 
-    let exit_status = child.wait().expect(
-        "failed to wait for cargo build child to finish",
-    );
+    let artifacts: Vec<_> = Message::parse_stream(&*stdout)
```

Comment:
Ah, I missed that this now duplicates the `Message::parse_stream()` call. In that case, it still makes sense to factor it out so that `!opt.dev` can share the same result.

Reply:
I'll leave that as is because the second `Message::parse_stream` is removed in this PR anyway.

Reply:
Ah, that's fine then.